mypyc/genops.py

```diff
@@ -1634,61 +1646,61 @@ def unbox_or_cast(self, src: Value, target_type: RType, line: int) -> Value:
     def box_expr(self, expr: Expression) -> Value:
         return self.box(self.accept(expr))
 
-    def load_from_outer_scope(self, symbol: SymbolNode) -> Optional[AssignmentTarget]:
-        """Searches for a SymbolNode in each of the environment's symtables.
+    def load_outer_env(self, base: Value, index: int, line: int) -> Value:
+        """Loads the environment class for a given base into a register.
 
-        Given a particular SymbolNode, traverses the environment stack starting with the topmost
-        environment, and searches for the AssignmentTarget associated with that SymbolNode in each
-        of the environment's symtables. If the SymbolNode and corresponding AssignmentTarget are
-        found, load the AssignmentTarget into a local register.
+        Additionally, iterates through all of the SymbolNodes in the symtable of the environment at
+        the given index, and maps those SymbolNodes to the register where the environment class was
+        loaded in a separate dictionary. This is done so that the register containing the
+        environment where any particular SymbolNode lives can be looked up easily.
 
-        Returns the AssignmentTarget associated with the SymbolNode, or None if it cannot be found.
+        Returns the register where the environment class was loaded.
         """
-        symbol_found = False
-        index = len(self.environments) - 2
-        while not symbol_found and index >= 1:
-            if symbol in self.environments[index].symtable:
-                symbol_found = True
-                rtype = self.environments[index].symtable[symbol].type
-            else:
-                index -= 1
+        env = self.add(GetAttr(base, ENV_ATTR_NAME, line))
+        for symbol in self.environments[index].symtable:
+            self.fn_infos[index].symbol_to_env[symbol] = env
```

Comment:
Could we skip building `symbol_to_env` and instead use this to directly construct the `symtable` of the inner function?